modules/ROOT/pages/migrate-client-apps.adoc

```diff
@@ -0,0 +1,149 @@
++
+Conversely, If the client credentials that you provide do not match, the request does not get associated with the client and fails.
+
+[[migrate_with_internal]]
+== Migrate Client Applications for APIs with Anypoint Platform Client Providers
+
+To migrate your client application:
+
+. Open a text editor and paste the following commands:
++
+[source,text,linenums]
+----
+curl --location --request POST 'https://anypoint.mulesoft.com/exchange/api/v1/organizations/:orgId/applications' \
+--header 'Authorization: Bearer <omitted>' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "clientId": ":myClientId",
+    "clientSecret": ":myClientSecret",
+    "grantTypes": [],
+    "name": "My Application Name",
+    "redirectUri": []
+}'
+----
+. In the command snippet, replace the `orgId` parameter with your organization Id and the content of the body with the appropriate application <<properties,properties>>. 
++
+[NOTE]
+You must use the organization ID of the master organization.
+. Run this command at the Command prompt.
+
+//[[properties]]
+//Kim, please do not edit the commented parts.
+// == Properties Reference for the Migrate Task
+
+// In the properties of the migrate command, provide values based on the following recommendations:
+
+// * `clientId`: <Your client ID>
+// * `clientSecret`: <Your client secret>
+// * (optional)`description`: <Informative description for the applications>
+// +
+// This property is displayed on the application page. 
+// * `grantTypes`: As no IDP is provided, leave an empty array 
+// * `name`: <Name of the application to be created>
+// +
+// This property is displayed when selecting an application before requesting contracts.
+// * `redirectUri`: <Array containing the redirect URIs for your application to use as an OAuth 2.0 client>
+// * (optional) `url`: <URL of the client application>
+// +
+// This property is displayed on the application page.
+
+[[migrate_with_external]]
+== Migrate Client Applications for APIs with External Client Providers
+
+When creating an application for an organization with registered external client providers, you can directly associate the application with the API that you want to consume. To reference an API, you must first retrieve the API instance ID, which is available from the API Manager UI.
+
+You can include the client ID and client secret when referencing an API. However, this only works if you using a PingFederate external client provider or to the Anypoint Platform client provider
+
+When creating applications referencing APIs associated with a provider, such as Okta or Keycloak that are registered using OpenID Dynamic registration, you cannot define client ID and client secret credentials. This is because the OpenID Connect Dynamic Client Registration specification does not include client ID and client secret when creating a client.
```

Comment:
If you use a PingFederate external client provider or the Anypoint Platform client provider, you can include the client ID and client secret when referencing an API. Otherwise, for providers like Okta or Keycloak, which are registered using OpenID Dynamic registration, you cannot define client ID and client secret credentials.

Reply:
Done! Thanks!